### DISCUSSION TITLE
Bonus collateral in decentralized stablecoin's liquidation function

### DISCUSSION THREAD

**Discussion by uddercover** [Cyfrin/foundry-full-course-cu#2388] [Q&A] [answered]
Maybe I missed it but, where does the bonus collateral (paid to the liquidator) come from? 
From what I understand, we are redeeming the collateral equivalent of DSC paid + the bonus collateral, all from the debtor's collateral balance. The issue with that is, the debtor's balance will always be short 10% i.e the bonus collateral given to the liquidator, and their healthFactor would never improve. 
Is this just an issue with my code?

**Comment by EngrPips** [comment, marked as answer]
The protocol is 200% collateralized, meaning before the user can mint $50 worth of `DSC` they must have deposited $100 worth of collateral. User healthFactor gets bad the moment the collateral backing their debt is not 2x their debt, so there is always extra collateral to pay the liquidator the bonus collateral. Do you mean when a liquidator liquidates a user that needs to be liquidated, the user healthFactor never gets improved?

**Reply by uddercover** [reply]
Thanks for your answer. I was working under the assumption of the debtor’s collateral value falling to less than 110% the osc they minted. At that point the system fails, so they can no longer be liquidated as their healthFactor never improves. Do you think this is a valid concern?

**Reply by EngrPips** [reply]
This is the primary reason the protocol was over-collateralized. This concern is not valid regarding the protocol if you think the 200% collateralization mechanism is a good way to avoid protocol illiquidity, but if you are trying to say the 200% collateralization is not a good mechanism and it will fail then you have a point, and then you need to suggest the mechanics which you think is better.

**Reply by uddercover** [reply]
Okay, that makes sense. Thanks for the explanation.

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
